modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/Modal.js

```diff
@@ -668,6 +668,13 @@ class Iframe extends React.Component {
 				iframeWindow,
 				processClose: this.props.processClose,
 			});
+
+			iframeWindow.onload = () => {
+				this.props.onOpen({
+					iframeWindow,
+					processClose: this.props.processClose,
+				});
+			};
```

Comment:
This is very surprising to me that it works! There should not be a difference between this `iframeWindow.onload` and [the parent](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/Modal.js#L678) `onLoad`. But there is! Why does our SPA trigger internal `onload`, but not external? Is this maybe some React quirk with iframes? :thinking: 

This code works, but `onload` definition within `onload` looks pretty cryptic to me. To make this clearer, we can maybe explicitly cover SPA navigation. Something like:
```js
if (iframeWindow.Liferay.SPA) {
	this.screenFlipHandler = iframeWindow.Liferay.on('screenFlip', () => {
		this.props.onOpen({
			iframeWindow,
			processClose: this.props.processClose,
		});
	});
}
```

Reply:
I'll work with whatever you guys think on this one. @markocikos are you suggesting that we make the explicit SPA change in addition to my change, or instead of?

Reply:
instead of

Reply:
I tested behavior with SPA off. Pagination triggers external `onLoad`, which executes `onOpen`, as expected. This issue is only on when SPA is turned on.

Reply:
@markocikos would you like me to apply the change and re-send? Or would you prefer to take over from here? Either is fine with me.

Reply:
I can take care of it, will mention you for review :slightly_smiling_face:

Reply:
Sounds great, thank you @markocikos!